### PR TITLE
[Content] Clarify use cases for buttons and links

### PR DESCRIPTION
**Pull request by sadiesaurus**
### WHY are these changes introduced?

This PR is to add clarification around the use cases for buttons and links. This is to help ensure that these components are used intentionally and consistently. 

Discussion related to this topic
- https://github.com/Shopify/polaris-ux/issues/147
- https://docs.google.com/document/d/1I8MHXoeCeg1AWReLbj5J8kwYjXU1eBaABM3eKF9K44w/edit#

### WHAT is this pull request doing?

- Add context to the Link component page
- Add context to the Button component page
- Add mention of the UnstyledLink component and when it should be used

Link component page changes:

The link component description

The link component best practices section

Button component page changes:

The button component description

The button component best practices section

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the [Polaris UI kit](https://polaris.shopify.com/resources/polaris-ui-kit)

### CODE REVIEW

**Comment by mirualves** [comment]
Awesome, @sadiesaurus, it's looking good :) What do you think are the next steps and who should we include in the conversation?

**Comment by sadiesaurus** [comment]
@mirualves Excellent! So from here, we need to determine what best practices we'd like to recommend for the unstyled link. In the meantime, I'll add the use case context to the button component page.

**Comment by kaelig** [comment]
Added myself as a reviewer so I can look at the markdown syntax before those changes get merged.

**Comment by sadiesaurus** [comment]
@ry5n 👋I'm wondering if you have thoughts and feels around what to recommend for best practices for UnstyledLink? Should we include any, or is it enough to inform folks that they should only use it if other variations of Link don't suit their needs?

**Comment by ry5n** [comment]
@sadiesaurus I think we should document the UnstyledLink component and encourage folks to use it as you have above, when the Link styles don’t meet their needs 👍

File: src/components/Button/README.md [hunk 1]
```diff
@@ -46,11 +48,23 @@ Buttons should:
   confusion and make merchants unsure of what to do next.
 - Be positioned in consistent locations in the interface.
 
+### Buttons versus links
+
+Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as "view shipping settings".
```
Comment (kaelig) [inline src/components/Button/README.md]:
```suggestion
Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as “view shipping settings”.
```

File: src/components/Button/README.md [hunk 2]
```diff
@@ -30,7 +30,9 @@ keywords:
 
 # Button
 
-Buttons are used to make common actions immediately visible and easy to perform with one click, tap, or keypress. Merchants can use them to navigate or to take action.
+Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as "view shipping settings".
+
+For navigational actions that appear within or directly following a sentence, use the [link component](/components/navigation/link).
```
Comment (kaelig) [inline src/components/Button/README.md]:
```suggestion
For navigational actions that appear within or directly following a sentence, use the [link component](https://polaris.shopify.com/components/navigation/link).
```

File: src/components/Link/README.md [hunk 3]
```diff
@@ -23,54 +23,35 @@ keywords:
...
 
----
-
-## Best practices
-
-Links should:
-
-- Consist of text that clearly describes either the action that merchants will take or the location they’ll navigate to.
-- Only be used in a sentence. For stand-alone navigational actions, [use the button component](https://polaris.shopify.com/components/actions/button).
+For actions that aren't related to navigation, use the [button component](/components/actions/button).
```
Comment (kaelig) [inline src/components/Link/README.md]:
```suggestion
For actions that aren’t related to navigation, use the [button component](https://polaris.shopify.com/components/actions/button).
```

File: src/components/Link/README.md [hunk 4]
```diff
@@ -23,54 +23,35 @@ keywords:
...
+## Best practices
 
-#### Do
+### Buttons versus links
 
-Learn more about <a>fulfilling orders</a> at the Shopify Help Center
+Links are used primarily for navigation, and usually appear within or directly following a sentence.
 
-#### Don’t
+Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as "view shipping settings".
```
Comment (kaelig) [inline src/components/Link/README.md]:
```suggestion
Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as “view shipping settings”.
```

File: src/components/Button/README.md [hunk 5]
```diff
@@ -30,7 +30,9 @@ keywords:
 
 # Button
 
-Buttons are used to make common actions immediately visible and easy to perform with one click, tap, or keypress. Merchants can use them to navigate or to take action.
+Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as "view shipping settings".
```
Comment (kaelig) [inline src/components/Button/README.md]:
```suggestion
Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as “view shipping settings”.
```

File: src/components/Button/README.md [hunk 6]
```diff
@@ -46,11 +48,23 @@ Buttons should:
   confusion and make merchants unsure of what to do next.
 - Be positioned in consistent locations in the interface.
 
+### Buttons versus links
+
+Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”. Plain buttons, which look similar to links, are used for less important or less commonly used actions, such as "view shipping settings".
+
+Links are used primarily for navigation, and usually appear within or directly following a sentence.
+
+The HTML for buttons and links carries specific meaning. Using these components intentionally and consistently results in:
```
Comment (kaelig) [inline src/components/Button/README.md]: Making the first part of this sentence clearly refer to React components could clarify the intent:

```suggestion
The markup resulting from `Button` and `Link` carries meaning. Using these components intentionally and consistently results in:
```

What do you think?

File: src/components/Link/README.md [hunk 7]
```diff
@@ -23,54 +23,35 @@ keywords:
...
 
-<a>Click here</a> to learn more about fulfilling orders
+The HTML for buttons and links carries specific meaning. Using these components intentionally and consistently results in:
```
Comment (kaelig) [inline src/components/Link/README.md]: Making the first part of this sentence clearly refer to React components could clarify the intent:

```suggestion
The markup resulting from `Button` and `Link` carries meaning. Using these components intentionally and consistently results in:
```

What do you think?

File: src/components/Link/README.md [hunk 8]
```diff
@@ -23,54 +23,35 @@ keywords:
...
-<!-- end -->
+- a more inclusive experience for assistive technology users
+- a more cohesive visual experience for sighted users
+- products that are easier to maintain at scale
 
-<!-- usagelist -->
+### Unstyled links
 
-#### Do
+If the existing link styles don't meet the needs of your project, then use the `UnstyledLink` component to create a custom link style.
```
Comment (kaelig) [inline src/components/Link/README.md]:
```suggestion
If the existing link styles don’t meet the needs of your project, then use the `UnstyledLink` component to create a custom link style.
```

File: src/components/Link/README.md [hunk 9]
```diff
@@ -23,54 +23,35 @@ keywords:
...
-- Orders
-
-#### Don’t
+---
 
-- To orders section
+## Content guidelines
 
-<!-- end -->
+The link component should follow the content guidelines for [links](/content/actionable-language#section-links).
```
Comment (kaelig) [inline src/components/Link/README.md]:
```suggestion
The link component should follow the content guidelines for [links](https://polaris.shopify.com/content/actionable-language#section-links).
```

File: src/components/Link/README.md [hunk 10]
```diff
@@ -119,7 +100,7 @@ Use for text links that point to a different website. They will open in a new br
 
 ## Related components
 
-- To create navigational actions that aren’t part of a line of text, [use the button component](https://polaris.shopify.com/components/actions/button)
+- For actions that don't appear within or directly following a sentence, use the use the [button component](/components/actions/button)
```
Comment (kaelig) [inline src/components/Link/README.md]:
```suggestion
- For actions that don’t appear within or directly following a sentence, use the [button component](https://polaris.shopify.com/components/actions/button)
```

File: src/components/Button/README.md [hunk 11]
```diff
@@ -277,8 +291,8 @@ Use when a button has been pressed and the associated action is in progress.
 
 ## Related components
 
-- To learn how to combine or lay out multiple buttons, [use the button group component](https://polaris.shopify.com/components/actions/button-group)
-- To embed an action into a line of text, [use the link component](https://polaris.shopify.com/components/navigation/link)
+- To combine or lay out multiple buttons, [use the button group component](/components/actions/button-group)
+- For navigational actions that appear within or directly following a sentence, use the use the [link component](/components/navigation/link)
```
Comment (kaelig) [inline src/components/Button/README.md]:
```suggestion
- For navigational actions that appear within or directly following a sentence, use the [link component](https://polaris.shopify.com/components/navigation/link)
```

File: src/components/Button/README.md [hunk 12]
```diff
@@ -277,8 +291,8 @@ Use when a button has been pressed and the associated action is in progress.
 
 ## Related components
 
-- To learn how to combine or lay out multiple buttons, [use the button group component](https://polaris.shopify.com/components/actions/button-group)
-- To embed an action into a line of text, [use the link component](https://polaris.shopify.com/components/navigation/link)
+- To combine or lay out multiple buttons, [use the button group component](/components/actions/button-group)
```
Comment (kaelig) [inline src/components/Button/README.md]:
```suggestion
- To combine or lay out multiple buttons, [use the button group component](https://polaris.shopify.com/components/actions/button-group)
```

**Pull request merged**